gym/lib/gym/generators/build_command_generator.rb

```diff
@@ -58,6 +58,8 @@ def actions
       def suffix
         suffix = []
         suffix << "CODE_SIGN_IDENTITY=#{Gym.config[:codesigning_identity].shellescape}" if Gym.config[:codesigning_identity]
+        suffix << "PROVISIONING=''" if Gym.config[:provisioning_profile_name].to_s.length > 0
```

Comment:
Yeah this allows the build to succeed, because you have to specify this one using the UUID. The empty string here effectively overrides the project setting with `Automatic`, while the actual provisioning profile you want to specify is provided at export time, specified by name.